src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerationsTests.cs

```diff
@@ -3125,20 +3144,889 @@ End Sub
+    Public Class Hoo
+        Public Sub Goo(j As IEnumerable(Of Integer))
+            Dim a = [|j|].DistinctBy(Function (p) p).Count()
+            [|j|].ElementAt(10)
+        End Sub
+    End Class
+End Namespace
+";
+            await VerifyVisualBasicAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestChunk()
+        {
+            var csharpCode = $@"
+using System.Collections.Generic;
+using System.Linq;
+
+public class Bar
+{{
+    public void Sub(IEnumerable<int> j)
+    {{
+        var a = [|j|].Chunk(1000).Count();
+        [|j|].ElementAt(10);
+    }}
+}}";
+            await VerifyCSharpAsync(csharpCode);
+
+            var vbCode = $@"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace Ns
+    Public Class Hoo
+        Public Sub Goo(j As IEnumerable(Of Integer))
+            Dim a = [|j|].Chunk(100).Count()
+            [|j|].ElementAt(10)
+        End Sub
+    End Class
+End Namespace
+";
+            await VerifyVisualBasicAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestTryGetNonEnumeratedCount()
+        {
+            var csharpCode = $@"
+using System.Collections.Generic;
+using System.Linq;
+
+public class Bar
+{{
+    public void Sub(IEnumerable<int> j)
+    {{
+        if (j.TryGetNonEnumeratedCount(out var count))
+        {{
+        }}
+
+        j.ElementAt(10);
+    }}
+}}";
+            await VerifyCSharpAsync(csharpCode);
+
+            var vbCode = $@"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace Ns
+    Public Class Hoo
+        Public Sub Goo(j As IEnumerable(Of Integer))
+            Dim count = 0
+            If j.TryGetNonEnumeratedCount(count) then
+
+            End If
+            j.ElementAt(10)
+        End Sub
+    End Class
+End Namespace
+";
+            await VerifyVisualBasicAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestByDefaultSupposeMethodNotEnuemratesArgument()
+        {
+            var csharpCode = @"
+using System.Collections.Generic;
+using System.Linq;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> j, IEnumerable<int> i)
+    {
+        j.UnionBy(i, p => p).ElementAt(10);
+        Method(j);
+    }
+
+    private void Method(IEnumerable<int> k)
+    {
+        foreach (var p in k) { }
+    }
+}";
+            await VerifyCSharpAsync(csharpCode);
+
+            var vbCode = @"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace Ns
+    Public Class Hoo
+        Public Sub Goo(j As IEnumerable(Of Integer), i As IEnumerable(Of Integer))
+            j.UnionBy(i, Function(p) p).ElementAt(10)
+            Method(j)
+        End Sub
+
+        Private Sub Method(j As IEnumerable(Of Integer))
+            For Each kk in j
+            Next
+        End Sub
+    End Class
+End Namespace
+";
+            await VerifyVisualBasicAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestEnumerationMethodFromEditorConfig()
```

Comment:
Ideally this test would have be `Theory` with editorconfig value as the test method parameter. You could then have:
```csharp

public async Task TestEnumerationMethodFromEditorConfig(string editorconfig)
{
  var arg = editorconfig.IsEmpty ? "j" : "[|j|]";
  var csharpCode = $@"
...
   Method1({arg});
        {arg}.UnionBy(i, p => p).ElementAt(10);
...
}";